{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/srayan/miniconda3/envs/idp/lib/python3.10/site-packages/pandas/core/arrays/masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).\n",
      "  from pandas.core import (\n",
      "/tmp/ipykernel_237732/1172264817.py:1: DeprecationWarning: \n",
      "Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),\n",
      "(to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)\n",
      "but was not found to be installed on your system.\n",
      "If this would cause problems for you,\n",
      "please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466\n",
      "        \n",
      "  import pandas as pd\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "import sys\n",
    "import gzip\n",
    "import shutil\n",
    "import json\n",
    "import re\n",
    "import argparse\n",
    "import requests\n",
    "import Bio.PDB as bpdb\n",
    "from Bio import Align\n",
    "import tmalign_mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprot_id = \"PLBL2_MOUSE\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprot_pdb_path = \"/nfs/turbo/lsa-tewaria/uniprot_df_small.csv\"\n",
    "uniprot_df = pd.read_csv(uniprot_pdb_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprot_json = tmalign_mapping.find_uniprot(uniprot_df, uniprot_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "main_sequence = tmalign_mapping.extract_sequence(uniprot_json)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          name  start  end  d_8046668\n",
      "0  3fgr_0A.pdb     62  238      False\n",
      "1  3fgr_0B.pdb    244  248      False\n",
      "2  3fgr_0C.pdb    249  592      False\n",
      "3  3fgt_0A.pdb     59  239      False\n",
      "4  3fgt_0B.pdb    249  592      False\n",
      "5  3fgw_0A.pdb     60  239      False\n",
      "6  3fgw_0B.pdb    244  594      False\n"
     ]
    }
   ],
   "source": [
    "domain_info = pd.read_csv(\"/nfs/turbo/lsa-tewaria/uniprot/PLBL2_MOUSE/domain_info.csv\", index_col=0)\n",
    "print(domain_info)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"/nfs/turbo/lsa-tewaria/uniprot/PLBL2_MOUSE/PLBL2_MOUSE.fasta\", \"w\") as f:\n",
    "    for x in range(len(domain_info[\"name\"])):\n",
    "        start = domain_info.iloc[x][\"start\"]\n",
    "        end = domain_info.iloc[x][\"end\"]\n",
    "        f.write(f\">{domain_info.iloc[x]['name']}\\n\")\n",
    "        f.write(main_sequence[start:end] + \"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprot_path = \"/nfs/turbo/lsa-tewaria/uniprot/PLBL2_MOUSE\"\n",
    "fasta_path = \"/nfs/turbo/lsa-tewaria/uniprot/PLBL2_MOUSE/PLBL2_MOUSE.fasta\"\n",
    "terminal_command = \"mmseqs easy-cluster \" + fasta_path + \" \" + os.path.join(uniprot_path, \"clusterRes\") + \" tmp --min-seq-id 0.8 -c 0.8 --cov-mode 1\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster /nfs/turbo/lsa-tewaria/uniprot/PLBL2_MOUSE/PLBL2_MOUSE.fasta /nfs/turbo/lsa-tewaria/uniprot/PLBL2_MOUSE/clusterRes tmp --min-seq-id 0.8 -c 0.8 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t13.45111\n",
      "Substitution matrix                 \tnucl:nucleotide.out,aa:blosum62.out\n",
      "Seed substitution matrix            \tnucl:nucleotide.out,aa:VTML80.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \t2147483647\n",
      "Alphabet size                       \tnucl:5,aa:21\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \t1\n",
      "Pseudo count b                      \t1.5\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t72\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.8\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Gap open cost                       \tnucl:5,aa:11\n",
      "Gap extension cost                  \tnucl:2,aa:1\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \ttrue\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \tnucl:0.200,aa:0.000\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Database type                       \t0\n",
      "Shuffle input database              \ttrue\n",
      "Createdb mode                       \t1\n",
      "Write lookup file                   \t0\n",
      "Offset of numeric ids               \t0\n",
      "\n",
      "createdb /nfs/turbo/lsa-tewaria/uniprot/PLBL2_MOUSE/PLBL2_MOUSE.fasta tmp/3867016085307960131/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \n",
      "\n",
      "Shuffle database cannot be combined with --createdb-mode 0\n",
      "We recompute with --shuffle 0\n",
      "Converting sequences\n",
      "[\n",
      "Time for merging to input_h: 0h 0m 0s 3ms\n",
      "Time for merging to input: 0h 0m 0s 3ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 32ms\n",
      "Create directory tmp/3867016085307960131/clu_tmp\n",
      "cluster tmp/3867016085307960131/input tmp/3867016085307960131/clu tmp/3867016085307960131/clu_tmp --max-seqs 20 -c 0.8 --cov-mode 1 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.8 --remove-tmp-files 1 \n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 1\n",
      "linclust tmp/3867016085307960131/input tmp/3867016085307960131/clu_tmp/928794620424525937/clu_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 72 --compressed 0 -v 3 --sub-mat nucl:nucleotide.out,aa:blosum62.out -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.8 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca 1 --pcb 1.5 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --gap-open nucl:5,aa:11 --gap-extend nucl:2,aa:1 --zdrop 40 --alph-size nucl:5,aa:13 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale nucl:0.200,aa:0.000 --adjust-kmer-len 0 --mask 0 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \n",
      "\n",
      "kmermatcher tmp/3867016085307960131/input tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref --sub-mat nucl:nucleotide.out,aa:blosum62.out --alph-size nucl:5,aa:13 --min-seq-id 0.8 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale nucl:0.200,aa:0.000 --adjust-kmer-len 0 --mask 0 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher tmp/3867016085307960131/input tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref --sub-mat nucl:nucleotide.out,aa:blosum62.out --alph-size nucl:5,aa:13 --min-seq-id 0.8 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale nucl:0.200,aa:0.000 --adjust-kmer-len 0 --mask 0 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 7 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 7 0s 33ms\n",
      "Sort kmer 0h 0m 0s 0ms\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\n",
      "Time for fill: 0h 0m 0s 0ms\n",
      "Time for merging to pref: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 167ms\n",
      "rescorediagonal tmp/3867016085307960131/input tmp/3867016085307960131/input tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref_rescore1 --sub-mat nucl:nucleotide.out,aa:blosum62.out --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 1 --min-seq-id 0.8 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 7 0s 18ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 190ms\n",
      "Time for processing: 0h 0m 0s 496ms\n",
      "clust tmp/3867016085307960131/input tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref_rescore1 tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 89ms\n",
      "\n",
      "Size of the sequence database: 7\n",
      "Size of the alignment database: 7\n",
      "Number of clusters: 3\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 128ms\n",
      "createsubdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/order_redundancy tmp/3867016085307960131/input tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 25ms\n",
      "createsubdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/order_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 26ms\n",
      "filterdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref_filter1 tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref_filter2 --filter-file tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/order_redundancy --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 3 0s 22ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 181ms\n",
      "Time for processing: 0h 0m 0s 444ms\n",
      "rescorediagonal tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/input_step_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/input_step_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref_filter2 tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref_rescore2 --sub-mat nucl:nucleotide.out,aa:blosum62.out --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 1 --min-seq-id 0.8 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 3 0s 18ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 171ms\n",
      "Time for processing: 0h 0m 0s 539ms\n",
      "align tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/input_step_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/input_step_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref_rescore2 tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/aln --sub-mat nucl:nucleotide.out,aa:blosum62.out -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.8 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca 1 --pcb 1.5 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --gap-open nucl:5,aa:11 --gap-extend nucl:2,aa:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 3 type: Aminoacid\n",
      "Target database size: 3 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 3 0s 1ms\n",
      "Time for merging to aln: 0h 0m 0s 10ms\n",
      "3 alignments calculated\n",
      "3 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 42ms\n",
      "clust tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/input_step_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/aln tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 134ms\n",
      "\n",
      "Size of the sequence database: 3\n",
      "Size of the alignment database: 3\n",
      "Number of clusters: 3\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 4ms\n",
      "Time for processing: 0h 0m 0s 157ms\n",
      "mergeclusters tmp/3867016085307960131/input tmp/3867016085307960131/clu_tmp/928794620424525937/clu_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pre_clust tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/clust --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 3 0s 33ms\n",
      "Clustering step 2\n",
      "[=================================================================] 3 0s 83ms\n",
      "Write merged clustering\n",
      "[=================================================================] 7 0s 307ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 155ms\n",
      "Time for processing: 0h 0m 0s 516ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref_filter1 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 5ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 6ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref_rescore1 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 88ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pre_clust -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 5ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/input_step_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 9ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/input_step_redundancy_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 4ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref_filter2 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 74ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/pref_rescore2 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 76ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/aln -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 6ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/linclust/2797593540505814195/clust -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 4ms\n",
      "createsubdb tmp/3867016085307960131/clu_tmp/928794620424525937/clu_redundancy tmp/3867016085307960131/input tmp/3867016085307960131/clu_tmp/928794620424525937/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 21ms\n",
      "prefilter tmp/3867016085307960131/clu_tmp/928794620424525937/input_step_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/input_step_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/pref_step0 --sub-mat nucl:nucleotide.out,aa:blosum62.out --seed-sub-mat nucl:nucleotide.out,aa:VTML80.out -s 1 -k 0 --k-score 2147483647 --alph-size nucl:5,aa:21 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 1 --comp-bias-corr 0 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca 1 --pcb 1.5 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 3 type: Aminoacid\n",
      "Estimated memory consumption: 977M\n",
      "Target database size: 3 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 3 0s 17ms\n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 3 0s 0ms\n",
      "Index statistics\n",
      "Entries:          298\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.000005\n",
      "Top 10 k-mers\n",
      "    RLDFDA\t1\n",
      "    CELNEA\t1\n",
      "    CNKNNA\t1\n",
      "    AVWNNA\t1\n",
      "    CECPNA\t1\n",
      "    IPFTNA\t1\n",
      "    NGYDQA\t1\n",
      "    PAGYQA\t1\n",
      "    WFYKRA\t1\n",
      "    YPQLRA\t1\n",
      "Time for index table init: 0h 0m 0s 723ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 3\n",
      "Target db start 1 to 3\n",
      "[================================================] =3 0s 1ms\n",
      "================\n",
      "2.617037 k-mers per position\n",
      "99 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 12ms\n",
      "Time for processing: 0h 0m 1s 8ms\n",
      "align tmp/3867016085307960131/clu_tmp/928794620424525937/input_step_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/input_step_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/pref_step0 tmp/3867016085307960131/clu_tmp/928794620424525937/aln_step0 --sub-mat nucl:nucleotide.out,aa:blosum62.out -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.8 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca 1 --pcb 1.5 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --gap-open nucl:5,aa:11 --gap-extend nucl:2,aa:1 --zdrop 40 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 3 type: Aminoacid\n",
      "Target database size: 3 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 3 0s 0ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 9ms\n",
      "3 alignments calculated\n",
      "3 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 28ms\n",
      "clust tmp/3867016085307960131/clu_tmp/928794620424525937/input_step_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/aln_step0 tmp/3867016085307960131/clu_tmp/928794620424525937/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 100ms\n",
      "\n",
      "Size of the sequence database: 3\n",
      "Size of the alignment database: 3\n",
      "Number of clusters: 3\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 111ms\n",
      "mergeclusters tmp/3867016085307960131/input tmp/3867016085307960131/clu tmp/3867016085307960131/clu_tmp/928794620424525937/clu_redundancy tmp/3867016085307960131/clu_tmp/928794620424525937/clu_step0 --threads 72 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 3 0s 23ms\n",
      "Clustering step 2\n",
      "[=================================================================] 3 0s 94ms\n",
      "Write merged clustering\n",
      "[=================================================================] 7 0s 402ms\n",
      "Time for merging to clu: 0h 0m 0s 227ms\n",
      "Time for processing: 0h 0m 0s 686ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/clu_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 73ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/input_step_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 9ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/input_step_redundancy_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 3ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/pref_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 16ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/aln_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 7ms\n",
      "rmdb tmp/3867016085307960131/clu_tmp/928794620424525937/clu_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 4ms\n",
      "createtsv tmp/3867016085307960131/input tmp/3867016085307960131/input tmp/3867016085307960131/clu tmp/3867016085307960131/cluster.tsv --threads 72 -v 3 \n",
      "\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 399ms\n",
      "Time for processing: 0h 0m 0s 736ms\n",
      "result2repseq tmp/3867016085307960131/input tmp/3867016085307960131/clu tmp/3867016085307960131/clu_rep --db-load-mode 0 --compressed 0 --threads 72 -v 3 \n",
      "\n",
      "[=================================================================] 3 0s 19ms\n",
      "Time for merging to clu_rep: 0h 0m 0s 377ms\n",
      "Time for processing: 0h 0m 0s 936ms\n",
      "result2flat tmp/3867016085307960131/input tmp/3867016085307960131/input tmp/3867016085307960131/clu_rep tmp/3867016085307960131/rep_seq.fasta --use-fasta-header -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 8ms\n",
      "createseqfiledb tmp/3867016085307960131/input tmp/3867016085307960131/clu tmp/3867016085307960131/clu_seqs --threads 72 -v 3 \n",
      "\n",
      "[=================================================================] 3 0s 32ms\n",
      "Time for merging to clu_seqs: 0h 0m 0s 370ms\n",
      "Time for processing: 0h 0m 0s 823ms\n",
      "result2flat tmp/3867016085307960131/input tmp/3867016085307960131/input tmp/3867016085307960131/clu_seqs tmp/3867016085307960131/all_seqs.fasta -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 21ms\n",
      "rmdb tmp/3867016085307960131/input -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 7ms\n",
      "rmdb tmp/3867016085307960131/input_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 5ms\n",
      "rmdb tmp/3867016085307960131/clu_seqs -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 78ms\n",
      "rmdb tmp/3867016085307960131/clu_rep -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 6ms\n",
      "rmdb tmp/3867016085307960131/clu -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 92ms\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.system(terminal_command)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_put = pd.read_csv(os.path.join(uniprot_path, \"clusterRes_cluster.tsv\"), sep=\"\\t\", header=None, names=[\"cluster\", \"pdb\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       cluster          pdb\n",
      "0  3fgr_0B.pdb  3fgr_0B.pdb\n",
      "1  3fgt_0A.pdb  3fgt_0A.pdb\n",
      "2  3fgt_0A.pdb  3fgr_0A.pdb\n",
      "3  3fgt_0A.pdb  3fgw_0A.pdb\n",
      "4  3fgw_0B.pdb  3fgw_0B.pdb\n",
      "5  3fgw_0B.pdb  3fgr_0C.pdb\n",
      "6  3fgw_0B.pdb  3fgt_0B.pdb\n"
     ]
    }
   ],
   "source": [
    "print(cluster_put)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "idp",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
